client/my-sites/domains/domain-management/dns/dns-add-new.jsx

```diff
@@ -119,16 +125,39 @@ class DnsAddNew extends React.Component {
 			},
 		} );
 
-		this.setFormState( this.formStateController.getInitialState() );
+		if ( this.props.recordToEdit ) {
+			this.loadRecord();
+		} else {
+			this.setFormState( this.formStateController.getInitialState() );
+		}
+	}
+
+	loadRecord() {
+		const { recordToEdit } = this.props;
+
+		const selectedDnsRecordFields = this.getFieldsForType( recordToEdit.type );
+		const recordAttributes = Object.keys( selectedDnsRecordFields ).reduce( ( obj, field ) => {
+			obj[ field ] = recordToEdit[ field ];
+			return obj;
+		}, {} );
```

Comment:
We'll need to check to see if this record is on the root domain here. Something like this maybe:
```suggestion
		const isRootDomainRecord = recordToEdit.domain === recordToEdit.name.replace( /\.$/, '' );
		const recordAttributes = Object.keys( selectedDnsRecordFields ).reduce( ( obj, field ) => {
			obj[ field ] = isRootDomainRecord && 'name' === field ? '' : recordToEdit[ field ];
			return obj;
		}, {} );
```

Reply:
Thanks for the review! I've created a method to process field values, as there were some other cases (`target` for SRV records, `data` in MX records) which also need some preprocessing.